coredns/plugin/handler.go

```diff
@@ -162,10 +166,13 @@ func (lh *Lighthouse) error(str string) error {
 	return plugin.Error(lh.Name(), errors.New(str)) // nolint:wrapcheck // Let the caller wrap it.
 }
 
-func (lh *Lighthouse) nextOrFailure(ctx context.Context, name string, w dns.ResponseWriter, r *dns.Msg, code int, err string) (int, error) {
+func (lh *Lighthouse) nextOrFailure(ctx context.Context, name string, state *request.Request, r *dns.Msg, rcode int) (int, error) {
```

Comment:
Since callers always pass `state.Name()`, we can eliminate the `name` parameter since we have the `state` instance.

Reply:
ok, it is solved